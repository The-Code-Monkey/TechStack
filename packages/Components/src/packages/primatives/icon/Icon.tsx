import React, { useContext, Suspense, memo } from 'react';
import { HelpCircle } from 'react-feather';

import { generateAutomationId, ConfigContext } from '../../utils';
import { BoxProps } from '../box';

import { getIcon } from './utils';
import { SvgWrapper } from './styled';

export interface Props extends Omit<BoxProps, 'children'> {
  name: string;
  noFill?: boolean;
}

const Icon = ({ autoid, name, ...rest }: Props) => {
  const config = useContext(ConfigContext);
  const Element = getIcon(config, name);

  return (
    <Suspense fallback={<HelpCircle />}>
      <SvgWrapper
        autoid={`${generateAutomationId(autoid ?? name)}_icon`}
        {...rest}
      >
        <Element />
      </SvgWrapper>
    </Suspense>
  );
};

export default memo(Icon);
