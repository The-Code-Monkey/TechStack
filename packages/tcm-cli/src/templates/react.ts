import { PackageJson } from 'type-fest';

import basicTemplate from './basic';
import { Template } from './template';

const reactTemplate: Template = {
  name: 'react',
  dependencies: [
    ...basicTemplate.dependencies,
    '@types/react',
    '@types/react-dom',
    'react',
    'react-dom',
    '@tsconfig/create-react-app',
  ],
  packageJson: {
    ...basicTemplate.packageJson,
    peerDependencies: {
      react: '>=16',
    },
    scripts: {
      ...basicTemplate.packageJson.scripts,
      test: 'tcm test --passWithNoTests',
    } as PackageJson['scripts'],
    jest: {
      testEnvironment: 'jsdom',
    },
  },
};

export default reactTemplate;
