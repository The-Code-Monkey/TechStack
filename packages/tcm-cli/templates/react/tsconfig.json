{
  // see https://www.typescriptlang.org/tsconfig to better understand tsconfigs
  "extends": "../../tsconfig.json",
  "include": ["src", "types"],
"exclude": ["src/**/__test__/*.test.ts*x"],
}
