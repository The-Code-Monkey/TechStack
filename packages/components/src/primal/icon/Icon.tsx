import { helpcircle as DefaultIcon } from '@aw-web-design/react-feather';
import type { IconTypes } from '@aw-web-design/react-feather';
import * as React from 'react';
const { memo, Suspense, lazy } = React;

import { SizesType } from '../../theme/types';
import { generateAutomationId } from '../../utils';

import { SvgWrapper } from './styled';

export interface Props {
  autoid?: string;
  name: IconTypes;
  noFill?: boolean;
  size?: keyof SizesType;
}

export const getIcon = (name: IconTypes) => {
  return lazy(() =>
    import(`@aw-web-design/react-feather`).then(module => ({
      default: module[name],
    }))
  );
};

export const FeatherIcon = ({ name, ...rest }: Props) => {
  const Element = getIcon(name);

  return (
    <Suspense fallback={<DefaultIcon {...rest} data-name='fallback-icon' />}>
      <Element />
    </Suspense>
  );
};

const Icon = ({ autoid, name, ...rest }: Props) => {
  return (
    <SvgWrapper
      autoid={`${generateAutomationId(autoid ?? name)}_icon`}
      {...rest}
    >
      <FeatherIcon name={name} />
    </SvgWrapper>
  );
};

export default memo(Icon);
