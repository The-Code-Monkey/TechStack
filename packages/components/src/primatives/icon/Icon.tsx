import FeatherIcon from '@aw-web-design/react-feather';
import * as React from 'react';
const { memo } = React;

import { generateAutomationId } from '../../utils';
import { BoxProps } from '../box';

import { SvgWrapper } from './styled';

export interface Props extends Omit<BoxProps, 'children'> {
  name: string;
  noFill?: boolean;
}

const Icon = ({ autoid, name, ...rest }: Props) => {
  // const config = useContext(ConfigContext);

  return (
      <SvgWrapper
        autoid={`${generateAutomationId(autoid ?? name)}_icon`}
        {...rest}
      >
       <FeatherIcon name={name} />
      </SvgWrapper>
  );
};

export default memo(Icon);
