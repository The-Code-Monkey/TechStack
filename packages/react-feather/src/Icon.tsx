import lazy from '@aw-web-design/react-lazy-named';
import * as React from 'react';

import { default as HelpCircle } from './icons/help-circle.js';
import { IconProps } from './types.js';

export interface Props extends IconProps {
  name: string;
}

const { Suspense, memo } = React;

export const getIcon = (name: string) => {
  return lazy(() => import(`./icons/${name}.js`));
};

export const Icon = ({ name, ...rest }: Props) => {
  const Element = getIcon(name);

  return (
    <Suspense fallback={<HelpCircle {...rest} data-name='fallback-icon' />}>
      <Element data-name={name} {...rest} />
    </Suspense>
  );
};

export default memo(Icon);
