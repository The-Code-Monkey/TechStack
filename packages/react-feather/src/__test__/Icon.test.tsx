import { render, waitFor, screen } from '@testing-library/react';
import jest from 'jest';
import React from 'react';

import { Icon } from '../Icon';

describe('<Icon />', () => {
  it('renders correctly', async () => {
    const { asFragment } = render(<Icon name='check' />);

    expect(asFragment()).toMatchSnapshot();

    await waitFor(() => expect(screen.getByTitle('check-icon')));

    expect(asFragment()).toMatchSnapshot();
  });
});
