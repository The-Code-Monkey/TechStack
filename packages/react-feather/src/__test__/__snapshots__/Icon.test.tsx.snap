// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> renders correctly 1`] = `
<DocumentFragment>
  <svg
    data-name="fallback-icon"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="12"
      cy="12"
      r="10"
    />
    <path
      d="M9.09 9a3 3 0 0 1 5.83 1c0 2-3 3-3 3"
    />
    <line
      x1="12"
      x2="12.01"
      y1="17"
      y2="17"
    />
  </svg>
</DocumentFragment>
`;

exports[`<Icon /> renders correctly 2`] = `
<DocumentFragment>
  <svg
    data-name="check"
    fill="none"
    height="24"
    stroke="currentColor"
    stroke-linecap="round"
    stroke-linejoin="round"
    stroke-width="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      check-icon
    </title>
    <polyline
      points="20 6 9 17 4 12"
    />
  </svg>
</DocumentFragment>
`;
